### PR TITLE
Partial refactor of template configuration.

### PR DESCRIPTION
**Pull request by seancpeters**
To eliminate JSON objects from low-level processing.
WORK IN PROGRESS
DO NOT MERGE

### CODE REVIEW

File: src/Microsoft.TemplateEngine.Core/Operations/ConditionalType.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Microsoft.TemplateEngine.Core.Operations
+{
+    // The commonly used conditional types. If more get added to ConditionalConfig.cs, they should be added here too.
+    public enum ConditionalType
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Core/Operations/ConditionalType.cs]: (No change required) The C-Style/XML-style evaluators will be provider-ified soon - if there are any considerations to take in to account for their configuration (which provider/any config/etc) we should keep them in mind

File: src/Microsoft.TemplateEngine.Core/Operations/ReplacementTokens.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.TemplateEngine.Core.Contracts;
+
+namespace Microsoft.TemplateEngine.Core.Operations
+{
+    public class ReplacementTokens : IReplacementTokens
+    {
+        public string Identity { get; private set; }
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Core/Operations/ReplacementTokens.cs]: nit: setters aren't required

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalConfig.cs [hunk 3]
```diff
@@ -47,5 +47,174 @@ public IEnumerable<IOperationProvider> Process(IComponentManager componentManage
...
+                    throw new Exception($"Unrecognized conditional type {style}");
+            }
+
+            return setup;
+        }
+
+        public static List<IOperationProvider> XmlConditionalSetup(string evaluatorType, bool wholeLine, bool trimWhiteSpace, string id)
+        {
+            // This is the operationId (flag) for the balanced nesting.
+            string commentFixingOperationId = "Fix pseudo comments";
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalConfig.cs]: These Ids should probably be generated on the off chance that we'd ever end up having multiple block comment type conditionals in the same file

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/FlagsConfig.cs [hunk 4]
```diff
@@ -38,5 +38,37 @@ public IEnumerable<IOperationProvider> Process(IComponentManager componentManage
                 yield return new SetFlag(flag, on, off, onNoEmit, offNoEmit, id, @default);
             }
         }
+
+        public static readonly string FlagConditionalSuffix = ":cnd";
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/FlagsConfig.cs]: public needed?

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/GlobalRunConfig.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
```
Comment (mlorbetske) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/GlobalRunConfig.cs]: nit: unused usings

File: src/dotnet-new3/Program.cs [hunk 6]
```diff
@@ -18,6 +18,8 @@ public class Program
     {
         public static int Main(string[] args)
         {
+            Console.ReadLine();
```
Comment (mlorbetske) [inline src/dotnet-new3/Program.cs]: We should watch out for adding this to commits

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/FlagsConfig.cs (hunk 4 above)
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/FlagsConfig.cs]: No, it's not needed. I thought I'd need them exposed for something else, but don't.

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalConfig.cs (hunk 3 above)
Comment (seancpeters) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Config/ConditionalConfig.cs]: I've changed the Id values so they're unique - added text indicating the style of conditional for each one.

**Pull request merged**